exercises/circular-buffer/src/test/java/CircularBufferTest.java

```diff
@@ -0,0 +1,162 @@
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class CircularBufferTest {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void readingFromEmptyBufferShouldThrowException() throws BufferIOException {
+        CircularBuffer<Integer> buffer = new CircularBuffer<>(1);
+
+        expectedException.expect(BufferIOException.class);
+        expectedException.expectMessage("Tried to read from empty buffer");
+        buffer.read();
+    }
+
+    @Ignore("Remove to run test")
+    @Test
+    public void canReadItemJustWritten() throws BufferIOException {
+        CircularBuffer<Integer> buffer = new CircularBuffer<>(1);
+
+        buffer.write(1);
+        assertEquals((int) buffer.read(), 1);
```

Comment:
I thought the same thing about the cast the first time I looked at it so I ran it and found out why. 

```
error: reference to assertEquals is ambiguous
        assertEquals(buffer.read(), 1);
        ^
  both method assertEquals(long,long) in Assert and method assertEquals(Object,Object) in Assert match
```